pkg/collector/py/datadog_agent.go

```diff
@@ -111,6 +117,115 @@ func LogMessage(message *C.char, logLevel C.int) *C.PyObject {
 	return C._none()
 }
 
+// GetSubprocessOutput runs the subprocess and returns the output
+//export GetSubprocessOutput
+func GetSubprocessOutput(argv **C.char, argc, raise int) *C.PyObject {
+
+	// https://github.com/golang/go/wiki/cgo#turning-c-arrays-into-go-slices
+	length := int(argc)
+	subprocessArgs := make([]string, length-1)
+	cmdSlice := (*[1 << 30]*C.char)(unsafe.Pointer(argv))[:length:length]
+	subprocessCmd := C.GoString(cmdSlice[0])
+	for i := 1; i < length; i++ {
+		subprocessArgs[i-1] = C.GoString(cmdSlice[i])
+	}
+	cmd := exec.Command(subprocessCmd, subprocessArgs...)
+
+	runtime.LockOSThread()
+	defer runtime.UnlockOSThread()
```

Comment:
let's make sure that we're in a different goroutine than the one that originally called the python check, otherwise this would unlock the goroutine from the OS thread too early